src/index.js

```diff
@@ -1,23 +1,29 @@
 import React from 'react'
 import { render } from 'react-dom'
+import replace from 'lodash/replace'
 import configureStore from './store/configureStore'
-import { configureAPI } from './actions/api'
+import { configureApp, createMountPoint } from './actions/api'
 import Root from './containers/Root'
 
-const renderApp = ({ endpoint }) => {
+const renderApp = ({ mountPoint, host, endpoint, fields, preview = false }) => {
+  createMountPoint(id)
+  configureApp(host, endpoint, fields, preview)
+
   const store = configureStore()
-  configureAPI(endpoint)
+  const id = replace(mountPoint, '#', '')
 
   render(
     <Root store={store} />,
-    document.getElementById('root')
+    document.getElementById(id)
```

Comment:
@yih-en 
Use this instead: 
https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelector